### PR TITLE
Add threat model

### PR DESCRIPTION
**Pull request by ferozsalam**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/master/USERS.md)
- [x] Thanks for contributing!

This PR documents Cilium's threat model, considering Cilium from the perspective of variety of attackers, and how these attackers might choose to compromise Cilium or the data that Cilium handles.

```release-note
docs: Document the threat model for Cilium
```

### CODE REVIEW

**Comment by ferozsalam** [comment]
@joestringer I've updated this document based on the feedback I received on the Google Doc, and converted it to reStructuredText. Let me know what you think!

File: Documentation/spelling_wordlist.txt
Comment (qmonnet) [inline Documentation/spelling_wordlist.txt]: This is probably unnecessary given that you have the lowercase version below. I believe lowercase entries are seen as case-insensitive by Sphinx.

File: Documentation/security/threat-model.rst
Comment (joestringer) [inline Documentation/security/threat-model.rst]: nit, wording:

```suggestion
This section presents a threat model for Cilium. This threat model
allows interested parties to understand:
```

Comment (joestringer) [reply Documentation/security/threat-model.rst]: (This is the general tone we try for in Cilium docs - state the facts about the content rather than including references to certain parties, including the authors).

File: Documentation/security/threat-model.rst
Comment (joestringer) [inline Documentation/security/threat-model.rst]: This line should be unnecessary, since the reference above already uses an internal reference.
```suggestion
```

File: Documentation/security/threat-model.rst
Comment (joestringer) [inline Documentation/security/threat-model.rst]: It's best to avoid full internal links in the tree, since we end up maintaining multiple versions, and this can inadvertently lead users who are reading one version of the docs into another different version of the docs and confuse them.

I'd suggest using internal anchors + `:ref:` instead - ie, apply this diff on top of the branch:

```
diff --git a/Documentation/overview/component-overview.rst b/Documentation/overview/component-overview.rst
index 3ef1f05e511a..7c8ce2809cf6 100644
--- a/Documentation/overview/component-overview.rst
+++ b/Documentation/overview/component-overview.rst
@@ -4,6 +4,8 @@
     Please use the official rendered version released here:
     https://docs.cilium.io
 
+.. _component_overview:
+
 ******************
 Component Overview
 ******************
diff --git a/Documentation/security/threat-model.rst b/Documentation/security/threat-model.rst
index 33b3fb0b8344..bfbe4b4788fe 100644
--- a/Documentation/security/threat-model.rst
+++ b/Documentation/security/threat-model.rst
@@ -30,11 +30,10 @@ against `the SLSA framework`_.
 
 In order to understand the following threat model, readers will need
 familiarity with basic Kubernetes concepts, as well as a high-level
-understanding of Cilium’s `architecture and components`_.
+understanding of Cilium’s :ref:`architecture and components<component_overview>`.
 
 .. _recent security audit: https://github.com/cilium/cilium.io/blob/main/Security-Reports/CiliumSecurityAudit2022.pdf
 .. _the SLSA framework:  https://slsa.dev/
-.. _architecture and components: https://docs.cilium.io/en/stable/overview/component-overview/
 
 Methodology
 -----------
```

File: Documentation/security/threat-model.rst
Comment (joestringer) [inline Documentation/security/threat-model.rst]:
```suggestion
This threat model considers eight different types of threat
```

File: Documentation/security/threat-model.rst
Comment (joestringer) [inline Documentation/security/threat-model.rst]:
```suggestion
For each threat actor, this guide uses the `the STRIDE methodology`_ to
assess likely attacks. Where one attack type in the STRIDE set can lead to others
```

File: Documentation/security/threat-model.rst
Comment (joestringer) [inline Documentation/security/threat-model.rst]: In this case we can use the imperative, to instruct the reader about how to think about the problem:
```suggestion
For ease of understanding, consider a single Kubernetes
```

File: Documentation/security/threat-model.rst
Comment (joestringer) [inline Documentation/security/threat-model.rst]: (Similar to previous case):

```suggestion
For the first scenario, consider an attacker who has been able to
```

File: Documentation/security/threat-model.rst
Comment (joestringer) [inline Documentation/security/threat-model.rst]: Does 'elsewhere' have any implications here or can we drop it?

```suggestion
   This will help in ensuring that Cilium is not starved of resources due to a misbehaving deployment in a cluster.
```

File: Documentation/security/threat-model.rst
Comment (joestringer) [inline Documentation/security/threat-model.rst]: I'm going back and forth a bit in my head about whether to consistently use the Kubernetes terminology in cases like below (which helpfully matches the diagrams), or switch to generic terminology that would help to reinforce that none of the underlying mechanisms here are Kubernetes-specific.

Example:

```suggestion
-  Non-“root” SSH or other console access to a node.
```

(I'm also fine with leaving this as-is, and if we want to improve this to be more generic in a subsequent set of changes we can do so after this PR)

File: Documentation/security/threat-model.rst
Comment (joestringer) [inline Documentation/security/threat-model.rst]: I don't know if there's a better way to represent this threat, but this paragraph and the paragraphs under `Cilium configuration` and `Cilium eBPF programs` seem like the same threat to me. Do you consider them as different?

Possible suggestion, feel free to adopt or ignore at your discretion: The table makes it look like there are three threats. We might be able to use some sort of marker (`†`?) for the first instance and just write `Same as above (†)` in the rows below.

File: Documentation/security/threat-model.rst
Comment (joestringer) [inline Documentation/security/threat-model.rst]: I believe at least Kubernetes refers to this as `privileged`:
```suggestion
-  Pods running with ``privileged: true`` or other significant
```

File: Documentation/security/threat-model.rst
Comment (joestringer) [inline Documentation/security/threat-model.rst]: I believe that this is accurate for the "root-equivalent Host attacker", though @jrfastab may have some ideas about ways to lock down & audit such cases as well.

(no change necessary at this time, just dropping an FYI)

File: Documentation/security/threat-model.rst
Comment (joestringer) [inline Documentation/security/threat-model.rst]: This text implies features that are not provided by Cilium:

```suggestion
| Observability    | TLS is required for all connectivity between      |
| data             | Cilium components, as well as for exporting       |
|                  | data to other destinations, removing the          |
|                  | scope for spoofing or tampering.                  |
```

File: Documentation/security/threat-model.rst
Comment (joestringer) [inline Documentation/security/threat-model.rst:452]: The equivalent previous sections used bulletpoints for this section, it would be good to be consistent.

File: Documentation/security/threat-model.rst
Comment (joestringer) [inline Documentation/security/threat-model.rst]: I think that this reference should be included, but it seems like it's already included below under `Recommended controls`. Do we need it here? The sentence reads a bit odd all by itself here.

File: Documentation/security/threat-model.rst
Comment (joestringer) [inline Documentation/security/threat-model.rst]: nit, consistency:
```suggestion
For such an attacker, our threat matrix is as follows:
```

File: Documentation/security/threat-model.rst
Comment (joestringer) [inline Documentation/security/threat-model.rst]: I think that this one is similar to the workload attacker entry, including the Cilium security benefits? Do you think there is a good way to cross-reference?

File: Documentation/security/threat-model.rst
Comment (joestringer) [inline Documentation/security/threat-model.rst]:
```suggestion
|                  | -  Creating a large number of CiliumIdentity      |
|                  |    resources could result in denial of service.   |
```

File: Documentation/security/threat-model.rst
Comment (joestringer) [inline Documentation/security/threat-model.rst]: Maybe this should also have a marker that indicates eBPF programs are also vulnerable for the `kubectl exec` case above, similar to my comments for the limited-privilege host attacker?

File: Documentation/security/threat-model.rst
Comment (joestringer) [inline Documentation/security/threat-model.rst]: I believe that there is also the possibility to influence routing of specific traffic and potentially introduce MITM a la https://github.com/cilium/cilium/security/advisories/GHSA-xr49-pqw6-848w .

File: Documentation/security/threat-model.rst
Comment (joestringer) [inline Documentation/security/threat-model.rst:661]: I wonder if there's also some etcd authorization configuration that's worth noting.

File: Documentation/security/threat-model.rst
Comment (joestringer) [inline Documentation/security/threat-model.rst:672]: I think it's worth noting somewhere in this section that typically we would expect users to configure an independent etcd instance for Cilium's management of the network vs. the etcd instance that is typically deployed as part of configuring a Kubernetes cluster. By separating these, we can mitigate some degree of cluster-wide impacts for attackers that gain access only to the Cilium etcd credentials.

File: Documentation/security/threat-model.rst
Comment (joestringer) [inline Documentation/security/threat-model.rst:673]: @rolinh may be interested to review this section.

**Review by joestringer — CHANGES_REQUESTED** [review]
Nice work! I have a scattering of comments, mostly minor nits with suggested code changes to improve. There's one or two comments below that might take a bit more than some wordsmithing.

File: Documentation/security/threat-model.rst
Comment (ferozsalam) [reply Documentation/security/threat-model.rst]: tgraf had a similar comment in the original document, so I think let's drop it from here.

File: Documentation/security/threat-model.rst
Comment (ferozsalam) [reply Documentation/security/threat-model.rst]: Yep, I think that suggestion makes sense – I don't consider them as a different threat, but wanted to make it clear that the configuration and eBPF programs were also affected.

Have updated to use the marker.

File: Documentation/security/threat-model.rst
Comment (ferozsalam) [reply Documentation/security/threat-model.rst]: 👍 makes sense, I had done this elsewhere but forgot here. Thanks for the spot!

File: Documentation/security/threat-model.rst
Comment (ferozsalam) [reply Documentation/security/threat-model.rst:661]: In our own docs the only suggestion we make is to [enable TLS](https://docs.cilium.io/en/stable/installation/k8s-install-external-etcd/#optional-configure-the-ssl-certificates) 

Is there anything in particular you were thinking of? I'm wary of recommending security controls that might not play nicely with Cilium if we haven't tested them.

File: Documentation/security/threat-model.rst
Comment (ferozsalam) [reply Documentation/security/threat-model.rst]: Have cross-linked!

File: Documentation/security/threat-model.rst
Comment (joestringer) [reply Documentation/security/threat-model.rst:661]: I don't think we have any guides, just generally wondering about the notion of RBAC with etcd. If we don't have any idea about that then we can leave this suggestion out for now.

**Review by joestringer — APPROVED** [review]
Looks good to me. I'd suggest squashing all of the commits together into one before merge (cf. https://medium.com/@slamflipstrom/a-beginners-guide-to-squashing-commits-with-git-rebase-8185cf6e62ec).

File: Documentation/security/threat-model.rst
Comment (qmonnet) [inline Documentation/security/threat-model.rst]: Nitpick: Can we please avoid curly apostrophes and curly quotes in the RST doc? I understand these are usually resulting from copy/pasting from Google Docs, but it always make things more painful whenever we need to grep or otherwise process the docs.

File: Documentation/security/threat-model.rst
Comment (qmonnet) [inline Documentation/security/threat-model.rst]: I'd drop `recent`, or add a date, because it's only recent at the time of your writing.

```suggestion
Cilium’s `security audit`_ assessed Cilium’s supply chain controls
```

File: Documentation/security/threat-model.rst
Comment (qmonnet) [inline Documentation/security/threat-model.rst]:
```suggestion
actors, placed at different parts of a typical deployment stack. We will
```

File: Documentation/security/threat-model.rst
Comment (qmonnet) [inline Documentation/security/threat-model.rst]: Consistency

```suggestion
-  network packets managed by Cilium
-  observability data collected by Cilium and stored by Hubble
```

File: Documentation/security/threat-model.rst
Comment (qmonnet) [inline Documentation/security/threat-model.rst]:
```suggestion
-  Container images should be regularly patched to reduce the chance of
   compromise.
-  Minimal container images should be used where possible.
-  Host-level privileges should be avoided where possible.
-  Ensure that the container users do not have access to the underlying
   container runtime.
```

File: Documentation/security/threat-model.rst
Comment (qmonnet) [inline Documentation/security/threat-model.rst]: Let's use a list like for the previous sections, for consistency?

```suggestion
- :ref:`gsg_encryption` should be configured to ensure the confidentiality of
  communication between workloads.
- TLS should be configured for communication between the Prometheus
  metrics endpoints and the Prometheus server.
- Network policies should be configured such that only the Prometheus
  server is allowed to scrape :ref:`Hubble metrics <metrics>` in particular.
```

File: Documentation/security/threat-model.rst
Comment (qmonnet) [inline Documentation/security/threat-model.rst]:
```suggestion
- :ref:`gsg_encryption` should be configured to ensure the confidentiality of
  communication between workloads.
```

File: Documentation/security/threat-model.rst
Comment (qmonnet) [inline Documentation/security/threat-model.rst:530]: No diagram? I liked them a lot for the previous subsections :grin: They help visualise the context, and even help reading the docs by separating each subsection.

Same questions for the remaining subsections below.

File: Documentation/security/threat-model.rst
Comment (qmonnet) [inline Documentation/security/threat-model.rst]: `This is not a precise term` -> what does `This` refer to?

File: Documentation/security/threat-model.rst
Comment (qmonnet) [inline Documentation/security/threat-model.rst]:
```suggestion
different types of privileges. This section is intended to cover any
```

File: Documentation/security/threat-model.rst
Comment (qmonnet) [inline Documentation/security/threat-model.rst]: No need for bold mark-up, here or for any other table header; it's already bold by default.

```suggestion
| Threat surface  | Identified STRIDE   | Cilium security benefits       |
|                 | threats             |                                |
```

File: Documentation/security/threat-model.rst
Comment (qmonnet) [inline Documentation/security/threat-model.rst]: Where does this asterisk point to? ([here](https://xkcd.com/2708/)? :) )
Please make it a link instead.

File: Documentation/security/threat-model.rst
Comment (qmonnet) [inline Documentation/security/threat-model.rst]: ... Same as above :)

File: Documentation/security/threat-model.rst
Comment (qmonnet) [inline Documentation/security/threat-model.rst]: Let's remove the dagger and use a link.

**Review by qmonnet — CHANGES_REQUESTED** [review]
Some formatting nits. Contents look good, as far as I can tell.

I note that the rendering for the lists in the array is not optimal (font size bigger than surrounding paragraphs), but it's not related to your PR. It's probably not worth spending time to address, either.

Thanks for this work!

File: Documentation/security/threat-model.rst
Comment (ferozsalam) [reply Documentation/security/threat-model.rst]: 👋 thanks for the feedback! I'm trying to point to the cell immediately above in this case. I tried using a link just now using the standard `:ref:` syntax, but something about my browser (Chrome)/the template/rst means that clicking on the link doesn't take you to the top of the cell containing the reference, but instead to a weird midpoint of the cell, where you can't actually see the linked text.

Can you think of a workaround or a better way of expressing the same point? I could switch it to 'Same as for the Cilium agent above.', perhaps?

File: Documentation/security/threat-model.rst
Comment (qmonnet) [inline Documentation/security/threat-model.rst]: This has not been updated, causing the GitHub workflow to complain:

```suggestion
.. _security audit: https://github.com/cilium/cilium.io/blob/main/Security-Reports/CiliumSecurityAudit2022.pdf
```

Comment (ferozsalam) [reply Documentation/security/threat-model.rst]: Thanks for the spot!

File: Documentation/security/threat-model.rst
Comment (qmonnet) [reply Documentation/security/threat-model.rst]: OK let's drop the links if they behave funny. “Same as for the Cilium agent” should work, or just “Same as above”, even. It's the `(*)` with no visible destination that I found confusing.

File: Documentation/security/threat-model.rst
Comment (ferozsalam) [reply Documentation/security/threat-model.rst:530]: I originally didn't think there was a neat way of representing the attacker in a diagram accurately for these threats, but gave it a go after this comment. I agree the consistency is better, although I'm still ambivalent about how accurate/intuitive the API server and Hubble attacker representations are. lmk what you think!

File: Documentation/security/threat-model.rst
Comment (ferozsalam) [reply Documentation/security/threat-model.rst]: I have removed that sentence, I think it made more sense in an older version of the document but agree that it wasn't clear (and didn't add anything) as it ended up in the PR.

File: Documentation/security/threat-model.rst
Comment (qmonnet) [reply Documentation/security/threat-model.rst:530]: Thanks, I think they look great!

File: Documentation/security/threat-model.rst
Comment (qmonnet) [reply Documentation/security/threat-model.rst]: Thanks for fixing the apostrophes, can we also have straight double quotes please? :pray:

Comment (ferozsalam) [reply Documentation/security/threat-model.rst]: Argh, of course. Done.

**Review by joestringer — APPROVED** [review]
(Approving changes to `CODEOWNERS`, I didn't take a fresh look at the rest)

**Review by qmonnet — APPROVED** [review]
All good from my side! But please squash your commits together.

Thanks a lot!

**Comment by joestringer** [comment]
I filed https://github.com/cilium/cilium/issues/24653 for the Travis failure, that's unrelated to this PR. Docs CI and reviewers are :heavy_check_mark: , there's been over a week for additional feedback. We can always iterate more, but this is a great basis for the threat model. Good to merge.

Thanks for your awesome work on this @ferozsalam !

**Comment by joestringer** [comment]
cc @pchaigno @brb could you take a fresh look at the threat model from a network security perspective?

**Pull request merged**
